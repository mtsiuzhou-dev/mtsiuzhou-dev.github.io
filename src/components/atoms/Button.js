import PropTypes from 'prop-types';
import style from './Button.module.scss';

const Button = ({
  isSubmit = false,
  text,
  className = '',
  onClick = null,
  isPressed = false,
  innerRef = null,
}) => (
  <button
    aria-pressed={isPressed}
    type={isSubmit ? 'submit' : 'button'}
    className={`${style.button} ${className || ''}`.trim()}
    onClick={onClick}
    ref={innerRef}
  >
    {text}
  </button>
);
Button.defaultProps = {
  isSubmit: false,
  onClick: null,
  isPressed: false,
  className: '',
  innerRef: null,
};
Button.propTypes = {
  isSubmit: PropTypes.bool,
  text: PropTypes.string.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func,
  isPressed: PropTypes.bool,
  innerRef: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({ current: PropTypes.instanceOf(Element) }),
  ]),
};
export default Button;
