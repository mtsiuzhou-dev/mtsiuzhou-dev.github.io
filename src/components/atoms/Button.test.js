import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('Button:', () => {
  const text = 'test';

  test('renders correctly', () => {
    const button = renderer.create(<Button text={text} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  test('onClick should be called', () => {
    const mockOnClick = jest.fn();
    render(<Button text={text} onClick={mockOnClick} />);
    userEvent.click(screen.getByText(text));

    expect(mockOnClick).toHaveBeenCalled();
  });
});
