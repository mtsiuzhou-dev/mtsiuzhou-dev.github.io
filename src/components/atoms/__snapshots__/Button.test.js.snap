// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button: renders correctly 1`] = `
<button
  aria-pressed={false}
  className="button"
  onClick={null}
  type="button"
>
  test
</button>
`;
